docs/using/permissions.md

```diff
@@ -0,0 +1,399 @@
+- Control access to sensitive credentials
+
+### Memory Permissions
+
+Set memory limits for components (future capability).
+
+**Example uses:**
+- Prevent resource exhaustion
+- Enforce quotas in multi-tenant environments
+
+## Granting Permissions
+
+The recommended way to grant permissions is through your AI agent when running Wassette as an MCP server. You can also use CLI commands for direct management, or define permissions in policy files.
+
+### Using MCP Built-in Tools (Recommended)
+
+When running Wassette as an MCP server, simply ask your AI agent to grant permissions in natural language:
+
+```text
+Please grant storage read and write permissions to the weather-tool for fs://workspace/
+```
+
+The agent will automatically use the appropriate built-in tool to apply the permission.
+
+**More examples:**
+
+```text
+Grant network access to api.weather.com for the weather-tool component
+```
+
+```text
+Allow the weather-tool to access the API_KEY environment variable
+```
+
+**Available MCP tools:**
+- `grant-storage-permission`: Grant file system access
+- `grant-network-permission`: Grant network access
+- `grant-environment-variable-permission`: Grant environment variable access
+
+The agent understands permission requests and selects the right tool, so you don't need to worry about command syntax.
+
+### Using CLI Commands
+
+For direct management or scripting, use the `wassette permission grant` command:
+
+**Grant storage access:**
+```bash
+# Read-only access to a directory
+wassette permission grant storage weather-tool fs://workspace/ --access read
+
+# Read and write access
+wassette permission grant storage weather-tool fs://workspace/ --access read,write
+
+# Access to a specific file
+wassette permission grant storage weather-tool fs://config/app.yaml --access read
+```
+
+**Grant network access:**
+```bash
+# Allow access to a specific host
+wassette permission grant network weather-tool api.weather.com
+
+# Allow localhost access
+wassette permission grant network weather-tool localhost:8080
+```
+
+**Grant environment variable access:**
+```bash
+# Grant access to an environment variable
+wassette permission grant environment-variable weather-tool API_KEY
+
+# Grant access to multiple variables
+wassette permission grant environment-variable weather-tool HOME
+wassette permission grant environment-variable weather-tool PATH
+```
+
+### Using Policy Files
+
+You can define permissions in a `policy.yaml` file alongside your component for pre-configured permissions:
+
+```yaml
+version: "1.0"
+description: "Weather tool permissions"
+permissions:
+  storage:
+    allow:
+      - uri: "fs://workspace/**"
+        access: ["read", "write"]
+      - uri: "fs://config/app.yaml"
+        access: ["read"]
+  network:
+    allow:
+      - host: "api.weather.com"
+      - host: "api.openweathermap.org"
+  environment:
+    allow:
+      - key: "API_KEY"
+      - key: "WEATHER_API_TOKEN"
+```
+
```

Comment:
Note that this is not meant to be manually write. It's a file that wassette can manipulate using the grant-* built in tools or CLI.